### PR TITLE
feat(guide): Port slash commands sections

### PR DESCRIPTION
**Pull request by monbrey**
This ports the Slash commands section of the guide to the new one.
Supersedes #9507  because I accidentally broke my old PR.

- Closes GDE-66
- Closes GDE-67
- Closes GDE-68
- Closes GDE-69
- Closes GDE-70
- Closes GDE-71
- Closes GDE-72

**Status and versioning classification:**
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+</CH.Code>
+
+If you have a command that performs longer tasks, be sure to call <DocsLink type="class" parent="ChatInputCommandInteraction" symbol="deferReply" brackets /> as early as possible.
+
+<Alert title="Warning" type="warning">
+	In these examples the _`wait`_ method is used to simulate a four second delay. You should not add uneccessary waits to
+	your actual commands.
+</Alert>
+
+Note that if you want your response to be ephemeral, you must pass an _`ephemeral`_ flag to the <DocsLink type="typedef" parent="InteractionDeferOptions" /> here:
```
Comment (Jiralite) [inline apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx:162]:
```suggestion
Note that if you want your response to be ephemeral, you must pass an _`ephemeral`_ flag to the <DocsLink type="typedef" parent="InteractionDeferReplyOptions" /> here:
```

File: apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+		Pong again!
+	</DiscordMessage>
+</DiscordMessages>
+
+Note that if you use <DocsLink type="class" parent="ChatInputCommandInteraction" symbol="followUp" brackets /> after a <DocsLink type="class" parent="ChatInputCommandInteraction" symbol="deferReply" brackets />, the first follow-up will edit the "_`<application> is thinking`_" message rather than sending a new one.
+
+That's all, now you know everything there is to know on how to reply to slash commands!
+
+<Alert title="Masked links" type="success">
+	Interaction responses can use masked links (e.g. `[text](http://site.com)`) in the message content.
```
Comment (Jiralite) [inline apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx:299]: Personally, I'd rather show `https://`:

```suggestion
	Interaction responses can use masked links (e.g. `[text](https://site.com)`) in the message content.
```

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+		},
+	],
+};
+```
+
+</CH.Code>
+
+## Option types
+
+By specifying the _`type`_ of an <DocsLink type="typedef" parent="ApplicationCommandOptionData" /> using the <DocsLink type="typedef" parent="ApplicationCommandOptionType" /> enum you are able to restrict what the user can provide as input, and for some options, leverage the automatic parsing of options into proper objects by Discord.
```
Comment (Jiralite) [inline apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:43]: `ApplicationCommandOptionType` is actually from discord-api-types.

**Review by Jiralite — CHANGES_REQUESTED** [review]

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx (hunk 3 above)
Comment (monbrey) [reply apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:43]: I agree, make DocsLink support it 😀

File: apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+```
+
+</CH.Code>
+
+Or, pass the _`fetchReply: true`_ option when sending the reply:
+
+<CH.Code lineNumbers={false}>
+
+```js
+const message = await interaction.reply(}content: 'Pong!', fetchReply: true });
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx:332]:
```suggestion
const message = await interaction.reply({ content: 'Pong!', fetchReply: true });
```

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+		{
+			name: 'input',
+			type: ApplicationCommandOptionType.String,
+			description :'The input to echo back',
+		},
+		{
+			name: 'ephemeral'
+			type: ApplicationCommandOptionType.Boolean,
+			description: 'Whether or not the echo should be ephemeral',
+		}
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:86]:
```suggestion
		},
```

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+	name: 'echo',
+	description: 'Replies with your input!',
+	options: [
+		{
+			name: 'input',
+			type: ApplicationCommandOptionType.String,
+			description :'The input to echo back',
+		},
+		{
+			name: 'ephemeral'
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:83]:
```suggestion
			name: 'ephemeral',
```

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+	name: 'echo',
+	description: 'Replies with your input!',
+	options: [
+		{
+			name: 'input',
+			type: ApplicationCommandOptionType.String,
+			description :'The input to echo back',
+		},
+		{
+			name: 'channel'
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:62]:
```suggestion
			name: 'channel',
```

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+```
+
+</CH.Code>
+
+Listed below is a short description of the different types of options that can be added.
+
+- _`String`_, _`Integer`_, _`Number`_ and _`Boolean`_ options all accept primitive values of their associated type.
+  - _`Integer`_ only accepts whole numbers.
+  - _`Number`_ accepts both whole numbers and decimals.
+- _`User`_, _`Channel`_, _`Role`_ and _`Mentionable`_ options will show a selection list in the Discord interface for their associated type, or will accept a Snowflake (id) as input.
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:98]:
```suggestion
- _`User`_, _`Channel`_, _`Role`_, and _`Mentionable`_ options will show a selection list in the Discord interface for their associated type, or will accept a Snowflake (id) as input.
```

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+		}
+	],
+};
+```
+
+</CH.Code>
+
+Listed below is a short description of the different types of options that can be added.
+
+- _`String`_, _`Integer`_, _`Number`_ and _`Boolean`_ options all accept primitive values of their associated type.
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:95]:
```suggestion
- _`String`_, _`Integer`_, _`Number`_, and _`Boolean`_ options all accept primitive values of their associated type.
```

File: apps/guide/src/content/04-slash-commands/03-parsing-options.mdx [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+<Alert title="Tip" type="success">
+If you want the Snowflake of a structure instead, grab the option via <DocsLink type="class" parent="CommandInteractionOptionResolver" symbol="get" brackets /> and access the Snowflake via the _`value`_ property. Note that you should use _`const { value: name } = ...`_ here to [destructure and rename](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) the value obtained from the <DocsLink type="typedef" parent="CommandInteractionOption" /> structure to avoid identifier name conflicts.
+</Alert>
+
+In the same way as the above examples, you can get values of any type using the corresponding <DocsLink type="class" parent="CommandInteractionOptionResolver" /> \_`get_____()`_ method. _`String`_, _`Integer`_, _`Number`_ and _`Boolean`_ options all provide the respective primitive types, while _`User`_, _`Channel`_, _`Role`_, and _`Mentionable`\_ options will provide either the respective discord.js class instance if your application has a bot user in the guild or a raw API structure for commands-only deployments.
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/03-parsing-options.mdx:60]: that looks really, really odd in the actual guide 
image

Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/03-parsing-options.mdx:60]:
```suggestion
In the same way as the above examples, you can get values of any type using the corresponding <DocsLink type="class" parent="CommandInteractionOptionResolver" /> \_`get_____()`_ method. _`String`_, _`Integer`_, _`Number`_ and _`Boolean`_ options all provide the respective primitive types, while _`User`_, _`Channel`_, _`Role`_, and _`Mentionable`_ options will provide either the respective discord.js class instance if your application has a bot user in the guild or a raw API structure for commands-only deployments.
```
assuming this should not be escaped?
image

File: apps/guide/src/content/04-slash-commands/05-autocomplete.mdx [hunk 11]
```diff
@@ -0,0 +1,250 @@
...
+
+<Alert title="Read first!" type="info">
+	This page is a follow-up to the [slash commands](./advanced-command-creation) section covering options and option
+	choices, and assumes you have also implemented [event handling](../creating-your-bot/event-handling). Please carefully
+	read those pages first so that you can understand the methods used in this section.
+</Alert>
+
+## Enabling autocomplete
+
+To use autocomplete with your commands, _instead_ of listing static choices, the option must be set to use autocompletion. This is done by setting `_autocomplete: true_` in the <DocsLink type="typedef" parent="ApplicationCommandOptionData" />. Autocomplete can be used on the same option types that support static choices; _`String`_, _`Integer`_ and _`Number`_.
```
Comment (almostSouji) [inline apps/guide/src/content/04-slash-commands/05-autocomplete.mdx:18]:
```suggestion
To use autocomplete with your commands, _instead_ of listing static choices, the option must be set to use autocompletion. This is done by setting _`autocomplete: true`_ in the <DocsLink type="typedef" parent="ApplicationCommandOptionData" />. Autocomplete can be used on the same option types that support static choices; _`String`_, _`Integer`_, and _`Number`_.
```
assuming this needs to be inverted - have not really worked with the new format yet
image

**Review by almostSouji — CHANGES_REQUESTED** [review]

File: apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx (hunk 4 above)
Comment (Jiralite) [reply apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx:332]: Interesting how Prettier doesn't fail for this.

File: apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx (hunk 3 above)
Comment (Jiralite) [reply apps/guide/src/content/04-slash-commands/02-advanced-command-creation.mdx:43]: Support added for this in 7b617bd22ecb4c009bdb0d807032134b51f6a855!

File: apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx [hunk 12]
```diff
@@ -0,0 +1,354 @@
...
+	await interaction.editReply('Pong again!');
+}
+```
+
+</CH.Code>
+
+If you have a command that performs longer tasks, be sure to call <DocsLink type="class" parent="ChatInputCommandInteraction" symbol="deferReply" brackets /> as early as possible.
+
+<Alert title="Warning" type="warning">
+	In these examples the _`wait`_ method is used to simulate a four second delay. You should not add uneccessary waits to
```
Comment (Jiralite) [inline apps/guide/src/content/04-slash-commands/01-command-response-methods.mdx:158]:
```suggestion
	In these examples the _`wait`_ method is used to simulate a four second delay. You should not add unnecessary waits to
```

**Comment by Jiralite** [comment]
Superseded by #10013.

**Pull request closed without merging**